sympy/stats/stochastic_process_types.py

```diff
@@ -759,3 +766,210 @@ def limiting_distribution(self):
+
+    >>> E(2*B[1] + B[2]).round(2)
+    2.10
+
+    >>> P(B[1] < 1).round(2)
+    0.30
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Bernoulli_process
+    .. [2] https://mathcs.clarku.edu/~djoyce/ma217/bernoulli.pdf
+
+    """
+
+    index_set = S.Naturals0
+
+    @property
+    def p(self):
+        return self.args[3]
+
+    @property
+    def distribution(self):
+        return self.args[2]
+
+    def __new__(cls, sym, p):
+        if p > 1 or p < 0:
+            raise ValueError("Probability must be in between 0 and 1")
+        p = sympify(p)
+        sym = _symbol_converter(sym)
+        state_space = _set_converter([0,1])
+        return Basic.__new__(cls, sym, state_space, BernoulliDistribution(p), p)
+
+    def probability_of_rth_success(self, ntrial, rsuccess):
+        """
+        Computes the probability of `rth` success in `nth` trail.
+
+        Parameters
+        ==========
+
+        ntrial: Positive Integer (ntrial > 1),
+          Represents the `nth` trial.
+        rsuccess: Positive Integer (rsuccess >= 1),
+          Represents `rth` success event (rsuccess <= ntrial).
+
+        Returns
+        =======
+
+        Probability that `rth` success occurs on `nth` trial.
+
+        """
+        if not (S(rsuccess).is_Integer and S(ntrial).is_Integer):
+            raise ValueError("nth-trial and rth-success must be Positive Integers.")
+
+        if not (rsuccess <= ntrial and rsuccess >= 1 and ntrial > 1):
+            raise ValueError("0 =< rth-success <= nth-trial, nth-trial > 1. ")
+
+        return (binomial(ntrial - 1, rsuccess - 1) * self.p**rsuccess
+                * (1 - self.p)**(ntrial - rsuccess))
+
+
+    def probability_r_success(self, ntrials, rsuccess, evaluate=True):
+        """
+        Computes the probability of `r` success events among the `n` trials.
+
+        Parameters
+        ==========
+
+        ntrials: Positive Integer,
+          Represents number of trials.
+        rsuccess: Positive Integer including 0, by default is None
+          Represents total number of success events (rsuccess <= ntrials).
+        evaluate: Boolean, by default is False
+          Computes the result numerically.
+
+        Returns
+        =======
+        Proability of getting `r` success in `n` trials
+            If evaluate is True
+        Distribution of getting `r` success in `n` trials
+            If evaluate is False
+
+        """
+        if evaluate:
+            if not (S(ntrials).is_Integer and S(rsuccess).is_Integer):
+                raise ValueError("ntrials and rsuccess must be postive Integers.")
+
+            if not (rsuccess <= ntrials and rsuccess >= 0):
+                raise ValueError("rsuccess <= ntrials and  ntrials, \
+                                     rsuccess must be positive Integers.")
+
+            return (binomial(ntrials, rsuccess) * self.p**rsuccess
+                    * (1 - self.p)**(ntrials - rsuccess))
+
+        else:
```

Comment:
Can we remove this `else`?. I think code will work fine without it.

Reply:
That else will work if evaluate is False in the argument

Reply:
Can you try removing it locally?

Reply:
If we want distribution at some particular value of `n` we will use it as 
`def probability_r_success(self, ntrials=5, rsuccess=None, evaluate=False):`